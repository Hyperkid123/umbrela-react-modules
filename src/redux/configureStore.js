import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
