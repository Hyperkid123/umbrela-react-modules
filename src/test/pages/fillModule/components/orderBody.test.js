import React from 'react'
import OrderBody, {OrderBody as Snapshot} from '../../../../pages/fillModule/components/orderBody';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Order body component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.answer = [];
    initialProps.createCards = jest.fn();
    initialProps.dragCard = jest.fn();
    initialProps.question = {
      questionId: 0,
      options: []
    };
    initialProps.option = {
      optionTitle: 'foo',
      optionId: 0
    };
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  const mockStore = configureStore();
  let initialState = {};
  let store;
  beforeEach(() => {
    initialState = {
      answers: {},
      locale: {
        languages: [{
          name: "English",
          code: 'en',
          active: true,
        }],
        translations: {}
      }
    }
  });

  it('should create component correctly', () => {
    store = mockStore(initialState);
    let wrapper = shallow(
      <OrderBody
        question={{
          questionId: 0
        }}
        store={store}
        option={{optionTitle: 'foo'}}
      />);
    expect(wrapper).toBeDefined();
  });

  it('should render correctly empty', () => {
    const dragCard = jest.fn();
    const wrapper = shallow(
      <Snapshot {...initialProps} dragCard={dragCard}/>
    )
    wrapper.instance().moveCard(10, 5);
    expect(dragCard).toHaveBeenCalledWith(0, [], 10, 5)
  })

  it('should call create cards action if no answer exists', () => {
    const createCards = jest.fn();
    const wrapper = shallow(
      <Snapshot {...initialProps} createCards={createCards} answer={false}/>
    )
    wrapper.instance().moveCard(10, 5);
    expect(createCards).toHaveBeenCalledWith(0, [])
  })

  it('should render options', () => {
    const answer = [{
      optionId: 0,
    }, {
      optionId: 1
    }, {
      optionId: 2,
    }]
    const tree = shallow(
      <Snapshot {...initialProps} answer={answer}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })
});
