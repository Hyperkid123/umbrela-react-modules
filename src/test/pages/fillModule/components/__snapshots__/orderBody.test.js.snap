// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order body component should render correctly empty 1`] = `<styled.div />`;

exports[`Order body component should render options 1`] = `
<styled.div>
  <DragSource(DropTarget(Card))
    id={0}
    index={0}
    key="0"
    moveCard={[Function]}
    option={
      Object {
        "optionId": 0,
      }
    }
  />
  <DragSource(DropTarget(Card))
    id={1}
    index={1}
    key="1"
    moveCard={[Function]}
    option={
      Object {
        "optionId": 1,
      }
    }
  />
  <DragSource(DropTarget(Card))
    id={2}
    index={2}
    key="2"
    moveCard={[Function]}
    option={
      Object {
        "optionId": 2,
      }
    }
  />
</styled.div>
`;
